src/utils/assets.js

```diff
@@ -332,10 +332,17 @@ export const isMatchingCollectible = (
 export const mapWalletAssetsBalancesIntoAssetsByAddress = (
   walletAssetsBalances: WalletAssetsBalances,
   chainSupportedAssets: Asset[],
-): AssetByAddress => mapValues(
-  walletAssetsBalances,
-  ({ address }: WalletAssetBalance) => findAssetByAddress(chainSupportedAssets, address),
-);
+): AssetByAddress => {
+  const walletAssets = Object.keys(walletAssetsBalances)
+    .map((assetAddress) => findAssetByAddress(chainSupportedAssets, assetAddress));
```

Comment:
Consider using `mapNotNil` from `utils/array` to remove not founds assets at this stage.

Reply:
Yeah, but then it's double iteration through array compared to `reduce` below which both builds the object and rejects assets that are not found 🤔